### ISSUE TITLE
Go e2e test flake

### ISSUE THREAD

**Issue by LiliC** [operator-framework/operator-sdk#1597]
I noticed a few times that our e2e test go operator type fails randomly, this happened recently in PRs that did not touch the go operator codebase.

For example, this just changes the helm codebase and the helm test, yet the go e2e test failed https://github.com/operator-framework/operator-sdk/pull/1594 

```
+operator-sdk test local ./test/e2e --up-local --namespace=test-memcached --kubeconfig /home/travis/.kube/config
[36mINFO[0m[0000] Testing operator locally.                    
time="2019-06-25T09:54:13Z" level=info msg="Started local operator"
--- FAIL: TestMemcached (80.13s)
    --- FAIL: TestMemcached/memcached-group (0.00s)
        --- FAIL: TestMemcached/memcached-group/Cluster2 (60.02s)
            memcached_test.go:109: Initialized cluster resources
            wait_util.go:48: Operator is running locally; skip waitForDeployment
            client.go:57: resource type  with namespace/name (test-memcached/example-memcached) created
            wait_util.go:64: Waiting for full availability of example-memcached deployment (0/3)
            wait_util.go:64: Waiting for full availability of example-memcached deployment (0/3)
            wait_util.go:64: Waiting for full availability of example-memcached deployment (0/3)
            wait_util.go:64: Waiting for full availability of example-memcached deployment (0/3)
            wait_util.go:64: Waiting for full availability of example-memcached deployment (0/3)
            wait_util.go:64: Waiting for full availability of example-memcached deployment (0/3)
            wait_util.go:64: Waiting for full availability of example-memcached deployment (0/3)
            wait_util.go:64: Waiting for full availability of example-memcached deployment (0/3)
            wait_util.go:64: Waiting for full availability of example-memcached deployment (0/3)
            wait_util.go:64: Waiting for full availability of example-memcached deployment (0/3)
            wait_util.go:64: Waiting for full availability of example-memcached deployment (0/3)
            wait_util.go:64: Waiting for full availability of example-memcached deployment (0/3)
            wait_util.go:64: Waiting for full availability of example-memcached deployment (0/3)
            memcached_test.go:123: failed waiting for 3 deployment/example-memcached replicas: timed out waiting for the condition
            client.go:75: resource type  with namespace/name (test-memcached/example-memcached) successfully deleted
FAIL
```

Nothing else in the logs to indicate what failed.

**Comment by AlexNPavel** [comment]
I have seen this occasionally. We need to print the operator's logs for more information on how to fix this. Could this have been related to the fix #1585?

In any case, we should be printing the operator logs

**Comment by LiliC** [comment]
> Could this have been related to the fix #1585?

Yes I think so as well, but the weird thing is that not even one deployment is ready. So extending the retry won't fix it I feel.
